File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        idx = len(nums)-1
        i = 0
        while i in range(len(nums)):
            if nums[i] == val:
                nums[i], nums[idx] = nums[idx], nums[i]
                idx -= 1
                i -= 1
            if i == idx: break
            i += 1
        return idx+1

File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int index = nums.size()-1;
        for(int i=0; i<nums.size(); i++){
            if(nums[i]==val) {
                swap(nums[i], nums[index]);
                index--;
                i--;
            }
            if(i==index) break;
        }
        return index+1;
    }
};